diff --git a/Cargo.toml b/Cargo.toml
index 20a700da..32ef9974 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,10 +13,11 @@ license = "Apache-2.0"
 panic = "abort"
 
 [profile.release]
-codegen-units = 1
-lto = true
+codegen-units = 16
+lto = false
 strip = true
 panic = "abort"
+opt-level = 2
 
 [patch.crates-io]
 #hiqlite = { git = "https://github.com/sebadob/hiqlite.git", branch = "optimized-migration-checks-in-writer", package = "hiqlite" }
